api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
```

Comment:
Yeah, I knew I missed something.
What about setting a filter function instead of using a condition.
```go
filterFunc := func(nw types.NetworkResource) bool {
    if danglingOnly {
        return !runconfig.IsPreDefinedNetwork(nw.Name) && len(nw.Containers) == 0
    }
    return runconfig.IsPreDefinedNetwork(nw.Name) || len(nw.Containers) > 0
}
...
for _, nw := range nws {
    if filterFunc(nw) {
        retNws = append(retNws, nw)
    }
}
```